lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -3880,16 +4291,48 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
     auto &adjoint = getAdjoint();
     auto adjLoc = getAdjoint().getLocation();
     LLVM_DEBUG(getADDebugStream() << "Running AdjointGen on\n" << original);
-    // Create entry BB and arguments.
-    auto *adjointEntry = adjoint.createBasicBlock();
+    auto *di = const_cast<DominanceInfo *>(&domInfo);
+    auto *pdi = const_cast<PostDominanceInfo *>(&postDomInfo);
+
+    // Create adjoint blocks and arguments.
+    {
+      PostDominanceOrder postDomOrder(&*original.findReturnBB(), pdi);
+      while (auto *origBB = postDomOrder.getNext()) {
+        auto *adjointBB = adjoint.createBasicBlock();
+        adjointBBMap.insert({origBB, adjointBB});
+        postDomOrder.pushChildren(origBB);
+        // If adjoint block is the adjoint entry, continue.
+        if (adjointBB->isEntry())
+          continue;
+        // Otherwise, add a primal value struct argument to the adjoint block.
+        auto pvStructLoweredType =
+            primalInfo.getPrimalValueStructLoweredType(origBB);
+        adjointBB->createPhiArgument(
+            pvStructLoweredType, ValueOwnershipKind::Guaranteed);
+      }
+    }
+
+    // Create primal value struct allocations for adjoint basic blocks with
```

Comment:
This code block (creating primal value struct allocations for adjoint basic blocks with multiple predecessors) may not be necessary. This was an experiment to work around verification errors from generating `switch_enum` terminators for adjoint bbs.

Delete this code block.